tools/utils/utils.js

```diff
@@ -457,9 +457,16 @@ exports.isUrlWithFileScheme = function (x) {
 };
 
 exports.isUrlWithSha = function (x) {
-  // For now, just support http/https, which is at least less restrictive than
-  // the old "github only" rule.
+  // Is a URL with a fixed SHA? We use this for Cordova -- although theoretically we could use
+  // a URL like isNpmUrl(), there are a variety of problems with this, 
+  // see https://github.com/meteor/meteor/pull/5562
   return /^https?:\/\/.*[0-9a-f]{40}/.test(x);
+}
+
+exports.isNpmUrl = function (x) {
+  // These are the various protocols that NPM supports, which we use to download NPM dependencies
+  // See https://docs.npmjs.com/files/package.json#git-urls-as-dependencies
+  return exports.isUrlWithSha(x) || /^(git|git\+ssh|git\+http|git\+https)?:\/\/.*#/.test(x);
```

Comment:
> Should we also allow URLs without a SHA for http(s)?

I guess npm says "You may specify a tarball URL in place of a version range".

But one nice property of the various `#commit-ish` URLs is that npm stores the actual commit hash in the shinkwrap file so you get consistent builds. If we allow arbitrary changeable[1] URLs, we lose that.

So I'd vote against it. Plus I don't recall people asking for it.

> Could we just add https? to the protocols instead of using isUrlWithSha?

No, because the `#commit-ish` thing is a property of `git+` URLs. (and the above)

> Now that we we allow more flexible URLs, is there any reason still to require an explicit branch instead of using the default one?

I'm just following npm's format here: https://docs.npmjs.com/files/package.json#git-urls-as-dependencies

[1] Ok in theory a URL that includes a hash is changeable but let's not get pathological.